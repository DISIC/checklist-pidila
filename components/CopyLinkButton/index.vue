<template>
  <li class="link-tool" v-on-click-outside="hideAlertSuccess">
    <p :role="displayedRoleAttribute" aria-live="assertive">
      <span v-if="displayedAlertSuccess" class="alert-success">
        Le lien a été copié dans le presse-papier.
      </span>
    </p>
    <button
      type="button"
      class="btn btn-tool btn-link"
      title="Copier le lien permanent vers cette liste"
      @click="copyUrl"
      @keyup.enter="copyUrl"
      @blur="hideAlertSuccess"
      @onfocusout="hideAlertSuccess"
      @keydown.esc="hideAlertSuccess">
      <svg class="svg-icon icon-tool icon-link click-through-able" aria-hidden="true" focusable="true">
        <use xlink:href="svg/icon-sprite.svg#link"></use>
      </svg>
      <span class="sr-only">Copier le lien permanent vers cette liste</span>
    </button>
  </li>
</template>

<script>
  export default {
    name: 'CopyLinkButton',
    data: function () {
      return {
        displayedAlertSuccess: false
      }
    },
    methods: {
      displayAlertSuccess: function () {
        this.displayedAlertSuccess = true
      },
      hideAlertSuccess: function () {
        this.displayedAlertSuccess = false
      },
      copyUrl: function (event) {
        var tempInput = document.createElement('input')
        tempInput.style = 'position: absolute; left: -9999px'
        tempInput.value = global.location.href
        tempInput.setAttribute('aria-hidden', 'true')
        tempInput.classList.add('sr-only')
        document.body.appendChild(tempInput)
        tempInput.select()
        document.execCommand('copy')
        document.body.removeChild(tempInput)
        event.target.focus()
        this.displayAlertSuccess()
      }
    },
    computed: {
      displayedRoleAttribute: function () {
        return this.displayedAlertSuccess ? '' : 'presentation'
      }
    }
  }
</script>
