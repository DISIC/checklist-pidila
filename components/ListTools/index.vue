<template>
  <ul class="list-tools">
    <copy-link-button/>
    <li class="print-tool">
      <button
        class="btn btn-tool btn-printer"
        type="button"
        title="Imprimer cette liste"
        onclick="window.print()">
        <svg class="svg-icon icon-tool icon-printer" aria-hidden="true" focusable="false">
          <use xlink:href="svg/icon-sprite.svg#printer"></use>
        </svg>
        <span class="sr-only">Imprimer cette liste</span>
      </button>
    </li>
    <li class="open-tool">
      <button @click="openAllIsPressed"
        class="btn btn-tool btn-open_all"
        title="Ouvrir toute la liste des critères"
        type="button">
        <svg class="svg-icon icon-tool icon-open_all" aria-hidden="true" focusable="false">
          <use xlink:href="svg/icon-sprite.svg#open_all"></use>
        </svg>
        <span class="sr-only">Ouvrir toute la liste des critères</span>
      </button>
    </li>
    <li class="close-tool">
      <button @click="closeAllIsPressed"
        class="btn btn-tool btn-close_all"
        title="Fermer toute la liste des critères"
        type="button">
        <svg class="svg-icon icon-tool icon-close_all" aria-hidden="true" focusable="false">
          <use xlink:href="svg/icon-sprite.svg#close_all"></use>
        </svg>
        <span class="sr-only">Fermer toute la liste des critères</span>
      </button>
    </li>
  </ul>
</template>

<script>
  import Events from '../../bus/events'
  import CopyLinkButton from '~/components/CopyLinkButton'

  export default {
    name: 'ListTools',
    components: {
      CopyLinkButton
    },
    data: function () {
      return {
        isDisplayed: true,
        currentUrl: '',
        hideTooltip: true
      }
    },
    methods: {
      openAllIsPressed: function () {
        this.$bus.$emit(Events.EXPAND_CRITERIA)
      },
      closeAllIsPressed: function () {
        this.$bus.$emit(Events.COLLAPSE_CRITERIA)
      }
    }
  }
</script>
