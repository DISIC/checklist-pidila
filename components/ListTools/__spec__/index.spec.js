import {shallowMount} from '@vue/test-utils'
import Events from '../../../bus/events'
import ListTools from '../index.vue'
import CopyLinkButton from '../../CopyLinkButton'

jest.useFakeTimers()

describe('components | ListTools', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(ListTools)

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit an "EXPAND_CRITERIA" event to the EventBus when "Ouvrir Toute la liste des critères" button is pressed', () => {
    // given
    const $bus = {}
    $bus.$emit = jest.fn()
    const wrapper = shallowMount(ListTools, {mocks: {$bus}})
    const expectedEventName = Events.EXPAND_CRITERIA

    // when
    wrapper.find('button.btn-open_all').trigger('click')

    // then
    expect($bus.$emit).toHaveBeenCalledWith(expectedEventName)
  })

  it('should emit an "COLLAPSE_CRITERIA" event to the EventBus when "Fermer Toute la liste des critères" button is pressed', () => {
    // given
    const $bus = {}
    $bus.$emit = jest.fn()
    const wrapper = shallowMount(ListTools, {mocks: {$bus}})
    const expectedEventName = Events.COLLAPSE_CRITERIA

    // when
    wrapper.find('button.btn-close_all').trigger('click')

    // then
    expect($bus.$emit).toHaveBeenCalledWith(expectedEventName)
  })

  it('should contain one CopyLinkButton component', () => {
    // when
    const wrapper = shallowMount(ListTools)

    // then
    const filterBoxComponents = wrapper.findAll(CopyLinkButton)
    expect(filterBoxComponents.wrappers).toHaveLength(1)
  })
})
