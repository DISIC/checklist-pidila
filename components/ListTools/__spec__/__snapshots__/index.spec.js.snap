// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | ListTools should match snapshot 1`] = `
<ul
  class="list-tools"
>
  <copy-link-button-stub />
   
  <li
    class="print-tool"
  >
    <button
      class="btn btn-tool btn-printer"
      onclick="window.print()"
      title="Imprimer cette liste"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-tool icon-printer"
        focusable="false"
      >
        <use
          xlink:href="svg/icon-sprite.svg#printer"
        />
      </svg>
       
      <span
        class="sr-only"
      >
        Imprimer cette liste
      </span>
    </button>
  </li>
   
  <li
    class="open-tool"
  >
    <button
      class="btn btn-tool btn-open_all"
      title="Ouvrir toute la liste des critères"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-tool icon-open_all"
        focusable="false"
      >
        <use
          xlink:href="svg/icon-sprite.svg#open_all"
        />
      </svg>
       
      <span
        class="sr-only"
      >
        Ouvrir toute la liste des critères
      </span>
    </button>
  </li>
   
  <li
    class="close-tool"
  >
    <button
      class="btn btn-tool btn-close_all"
      title="Fermer toute la liste des critères"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-icon icon-tool icon-close_all"
        focusable="false"
      >
        <use
          xlink:href="svg/icon-sprite.svg#close_all"
        />
      </svg>
       
      <span
        class="sr-only"
      >
        Fermer toute la liste des critères
      </span>
    </button>
  </li>
</ul>
`;
